src/pip/_internal/download.py

```diff
@@ -609,6 +611,12 @@ def __init__(self, *args, **kwargs):
     def add_insecure_host(self, host):
         # type: (str) -> None
         self.mount('https://{}/'.format(host), self._insecure_adapter)
```

Comment:
Since you defined `build_url_from_netloc()`, you might as well use it above, too.